pipeline_tools/input_utils.py

```diff
@@ -1,335 +1,168 @@
-def get_optimus_lanes(metadata_json, version):
-    """Get the lane metadata
+def get_urls_to_files_for_ss2(bundle):
+    """Return the direct urls to the input fastq files for SmartSeq2 pipeline.
 
     Args:
-        metadata_json (dict): metadata
-        version (str): the metadata version
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
 
     Returns:
-        Dict of lane metadata 
-
-    Raises:
-        NotImplementedError: if metadata version is unsupported
+         tuple: A tuple consisting of the url to the input fastq files, which are corresponding to read index 1 and 2
+         respectively.
     """
-    if version.startswith('4.'):
-        return _get_optimus_lanes_v4(metadata_json)
-    else:
-        raise NotImplementedError('Only implemented for v4 metadata')
+    fastq_1_url = fastq_2_url = None
+    sequence_files = bundle.sequencing_output
 
+    for sf in sequence_files:
+        if fastq_1_url and fastq_2_url:
+            return fastq_1_url, fastq_2_url  # early termination
+        if sf.read_index == 'read1':
+            fastq_1_url = sf.manifest_entry.url
+        if sf.read_index == 'read2':
+            fastq_2_url = sf.manifest_entry.url
+    return fastq_1_url, fastq_2_url
 
-def _get_optimus_lanes_v4(assay_json):
-    """Return the lane metadata from the assay json"""
-    lanes = assay_json['content']['seq']['lanes']
-    return lanes
 
+def download_file(item, dss_url, http_requests=HttpRequests()):
+    """Download the metadata for a given bundle from the HCA data store (DSS).
 
-def get_optimus_inputs(lanes, manifest_files):
-    """Returns metadata for optimus input files
+    This function borrows a lot of existing code from the `metadata-api` code base for consistency,
+    and this won't be required after migrating to use the HCA DSS Python API `dss_client` directly.
 
     Args:
-        lanes (dict): lane metadata
-        manifest_files (dict): file metadata
-
-    Returns:
-        Three lists of urls, representing fastqs for r1, r2, and i1, respectively.
-    In each list, the first item is for the first lane, the second item is for the second lane, etc.
+        item (typing.ItemsView): A dictionary's ItemsView object consisting of file_name and the manifest_entry.
+        dss_url (str): The url for the DCP Data Storage Service.
+        http_requests (HttpRequests): The HttpRequests object to use.
     """
-    r1 = [manifest_files['name_to_meta'][lane['r1']]['url'] for lane in lanes]
-    r2 = [manifest_files['name_to_meta'][lane['r2']]['url'] for lane in lanes]
-    i1 = [manifest_files['name_to_meta'][lane['i1']]['url'] for lane in lanes]
+    file_name, manifest_entry = item
+    file_uuid = manifest_entry['uuid']
+    return file_name, dcp_utils.get_file_by_uuid(file_id=file_uuid, dss_url=dss_url, http_requests=http_requests)
 
-    return r1, r2, i1
 
+def get_metadata_files(metadata_files_dict, dss_url, num_workers=None):
```

Comment:
This can be for future work, but there should be a way for us to get the bundle with the file metadata included in it already. Then once the bundles are plate based and include many more files, we won't have to fetch each one individually!

Reply:
Good point! Actually it is already there: https://github.com/HumanCellAtlas/metadata-api/blob/develop/examples/simple.ipynb 

As I said in the PR's description, we cannot use that approach(dss_client.download_bundle_metadata) because we have the `HttpRequests` class which helps us log the requests and retry on various failures. After discussing with @dshiga I think it is better we leave it as a future work and refactor the pipeline-tools to make it more manageable and efficient.